packages/core/src/NavigationContainer.tsx

```diff
@@ -86,18 +87,25 @@ const getPartialState = (
  */
 const Container = React.forwardRef(function NavigationContainer(
   { initialState, onStateChange, children }: Props,
-  ref: React.Ref<NavigationHelpers<ParamListBase>>
+  ref
 ) {
   const [state, setNavigationState] = React.useState<State>(() =>
     getPartialState(initialState)
   );
 
+  const {
+    listeners,
+    addNavigationListener,
+    removeNavigationListener,
+  } = useFocusedListeners();
+
   const dispatch = (action: NavigationAction) =>
-    context.performTransaction(() => {
-      actionListeners[0](action, undefined, null);
-    });
+    listeners[0](navigation => navigation.dispatch(action));
 
   React.useImperativeHandle(ref, () => ({
+    performOnFocusedNavigator: (
```

Comment:
Don't think we want to expose `performOnFocusedNavigator`. The exposed ref should look like every other navigation object. We should expose `canGoBack` method instead where we internally perform this on focused navigator and return the result.